### PR TITLE
MongoDb Schema Describer

### PR DESCRIPTION
**Pull request by pimeys**
Unifying the index fetching for IE and ME. It is way too complex to do twice...

### CODE REVIEW

File: introspection-engine/connectors/mongodb-introspection-connector/src/lib.rs [hunk 1]
```diff
@@ -59,6 +61,37 @@ impl MongoDbIntrospectionConnector {
...
     }
+
+    async fn describe(&self, preview_features: BitFlags<PreviewFeature>) -> ConnectorResult<MongoSchema> {
+        let mut schema = mongodb_schema_describer::describe(&self.connection, &self.database)
+            .await
+            .map_err(crate::Error::from)?;
+
+        if !preview_features.contains(PreviewFeature::FullTextIndex) {
+            #[allow(clippy::needless_collect)] // well, mr. clippy, maybe you should read about the borrow checker...
+            let kept_indexes: Vec<_> = schema.drain_indexes().filter(|i| !i.is_fulltext()).collect();
```
Comment (tomhoule) [inline introspection-engine/connectors/mongodb-introspection-connector/src/lib.rs:72]: Doesn't that invalidate index ids?

**Review by tomhoule — COMMENTED** [review]
Just starting to review, but I think there's a problem.

File: introspection-engine/connectors/mongodb-introspection-connector/src/lib.rs (hunk 1 above)
Comment (pimeys) [reply introspection-engine/connectors/mongodb-introspection-connector/src/lib.rs:72]: No, we only have them in one collection, and then keep the indexes in a vector. This empties both:

https://github.com/prisma/prisma-engines/pull/2449/files#diff-cf78c2b811a9b0c673afa5262b4852b1dc00ab5d1b4ca84fc31c5599d427c37eR132-R135

**Review by tomhoule — APPROVED** [review]
We had a zoom call with @do4gr and @pimeys and it's good to merge :+1:

**Pull request merged**